pkg/driver/driver.go

```diff
@@ -60,9 +74,19 @@ func NewDriver(options ...func(*DriverOptions)) (*Driver, error) {
 	}
 
 	driver := Driver{
-		controllerService: newControllerService(&driverOptions),
-		nodeService:       newNodeService(),
-		options:           &driverOptions,
+		options: &driverOptions,
+	}
+
+	switch driverOptions.mode {
```

Comment:
We probably should selectively register the gRPC service for each mode as well at `driver.Run()`:

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/9ca95cea4aaba3b03ae5346520070e629509f748/pkg/driver/driver.go#L94-L96